manifests/service.pp

```diff
@@ -9,26 +9,61 @@
   $refresh_systemd       = $confluence::params::refresh_systemd,
 ) {
 
-  if($refresh_systemd) {
-    include ::systemd::systemctl::daemon_reload
-  }
+  if $confluence::systemd_module == 'camptocamp' {
+    if($refresh_systemd) {
+      include ::systemd::systemctl::daemon_reload
+    }
+
+    file { $service_file_location:
```

Comment:
this file+service resource could be heavily reduced by using the https://github.com/camptocamp/puppet-systemd/#unit-files